checker/checker.go

```diff
@@ -289,8 +289,15 @@ func (c *checker) resolveOverload(
 		if len(overload.TypeParams) > 0 {
 			// Instantiate overload's type with fresh type variables.
 			substitutions := newMapping()
+			// When a type parameter within an argument bears the same name as another
+			// type parameter in the list it should not be added as an alternative
+			// variable subsitution for the overload result type.
```

Comment:
I'm confused about how this could happen.  For instance, even for `+: (list<A>, list<A>) --> A`, the type param list is just "A".  Please add to comment explaining how we can wind up in this situation.

Reply:
You're right, this was totally unnecessary.